gtsam/geometry/Pose3.cpp

```diff
@@ -63,6 +63,92 @@ Matrix6 Pose3::AdjointMap() const {
   return adj;
 }
 
+/* ************************************************************************* */
+// Calculate AdjointMap applied to xi_b, with Jacobians
+Vector6 Pose3::Adjoint(const Vector6& xi_b, OptionalJacobian<6, 6> H_this,
+                       OptionalJacobian<6, 6> H_xib) const {
+  //  Ad   *   xi   =   [   R   0    .  [w
+  //                      [t]R  R ]      v]
+  // Declarations, aliases, and intermediate Jacobians easy to compute now
+  Vector6 result;
+  auto Rw = result.head<3>();
+  const Vector3 &w = xi_b.head<3>(), &v = xi_b.tail<3>();
+  Matrix3 Rw_H_R, Rv_H_R;
+  const Matrix3 &Rw_H_w = R_.matrix();
```

Comment:
Matrix() is expensive when quaternion so do only once

Reply:
Intermediate things should not be references, that is an argument thing

Reply:
Oops thanks, didn't notice the quaternion case - done.